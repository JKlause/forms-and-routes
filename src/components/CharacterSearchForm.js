import React from 'react';
// import styles from './Footer.css';
import PropTypes from 'prop-types';

const CharacterSearchForm = ({ handleSubmit, handleChange, searchQuery }) => {
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" name="searchQuery" placeholder="Search For A Character" value={searchQuery} onChange={handleChange}></input>
      <button>Search</button>
    </form>
  );
};

CharacterSearchForm.propTypes = {
  searchQuery: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired
};


export default CharacterSearchForm;
