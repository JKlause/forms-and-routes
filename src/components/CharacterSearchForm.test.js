import React from 'react';
import { shallow } from 'enzyme';
import CharacterSearchForm from './CharacterSearchForm';

describe('CharacterSearchForm component', () => {

  const props = {
    searchQuery: '',
    handleSubmit: () => {},
    handleChange: () => {}
  };
  it('renders CharacterSearchForm', () => {   
    const wrapper = shallow(<CharacterSearchForm {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
