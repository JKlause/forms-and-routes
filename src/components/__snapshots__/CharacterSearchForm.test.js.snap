// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterSearchForm component renders CharacterSearchForm 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="searchQuery"
    onChange={[Function]}
    placeholder="Search For A Character"
    type="text"
    value=""
  />
  <button>
    Search
  </button>
</form>
`;
